apps/scratch/src/utilities/student-activity-tracking.ts

```diff
@@ -1,61 +1,76 @@
 import { CrtContextValue } from "../contexts/CrtContext";
 import { isBlockPartOfLargeStack } from "./scratch-block-stack-utils";
-import type { Block, Workspace } from "scratch-blocks";
+import type { Block } from "scratch-blocks";
 import type { WorkspaceChangeEvent } from "../types/scratch-workspace";
 interface TrackMoveParams {
-  workspace: Workspace;
-  blockId?: string;
-  sendRequest?: CrtContextValue["sendRequest"];
-  action?: StudentAction;
+  block: Block;
+  sendRequest: CrtContextValue["sendRequest"];
+  action: StudentAction;
+  solution: Blob;
 }
 
-export type StudentAction = "create" | "move";
+export enum StudentAction {
+  CREATE = "create",
+  MOVE = "move",
+  DELETE = "delete",
+}
+
+const scratchToStudentAction: Record<string, StudentAction | null> = {
+  create: StudentAction.CREATE,
+  move: StudentAction.MOVE,
+  delete: StudentAction.DELETE,
+};
 
-export function shouldTrackActivity(
+export const mapScratchEventTypeToStudentAction = (
+  type: string,
+): StudentAction | null => scratchToStudentAction[type] || null;
+
+export const shouldTrackActivity = (
+  action: StudentAction,
   event: WorkspaceChangeEvent,
   canEditTask: boolean | undefined,
-): boolean | undefined {
-  return (
-    (event.type === "move" || event.type === "create") &&
-    event.recordUndo &&
-    !canEditTask &&
-    !!event.blockId
-  );
-}
+): boolean | undefined =>
+  (action === StudentAction.MOVE || action === StudentAction.CREATE) &&
+  event.recordUndo &&
```

Comment:
I am still confused by this here and would advice documenting what the purpose of this check (and the other checks) is. Also the function is named `shouldTrackMove`. It may be worth renaming this to what we want to do, not what the underlying scratch data types are called.